mdal/frmts/mdal_driver.cpp

```diff
@@ -87,7 +87,12 @@ void MDAL::Driver::createDataset( MDAL::DatasetGroup *group, double time, const
 {
   std::shared_ptr<MDAL::MemoryDataset> dataset = std::make_shared< MemoryDataset >( group );
   dataset->setTime( time );
-  memcpy( dataset->values(), values, sizeof( double ) * dataset->valuesCount() );
+  size_t count = dataset->valuesCount();
+
+  if ( !group->isScalar() )
+    count *= 2;
```

Comment:
Please check if this is OK, without this only half of the array gets copied for vector data.

Reply:
it is ok